pkg/farm/list_builder.go

```diff
@@ -0,0 +1,320 @@
+package farm
+
+import (
+	"context"
+	"fmt"
+	"io/fs"
+	"os"
+	"path/filepath"
+	"sync"
+
+	lmanifests "github.com/containers/common/libimage/manifests"
+	"github.com/containers/common/pkg/supplemented"
+	cp "github.com/containers/image/v5/copy"
+	"github.com/containers/image/v5/manifest"
+	"github.com/containers/image/v5/signature"
+	"github.com/containers/image/v5/transports/alltransports"
+	"github.com/containers/image/v5/types"
+	"github.com/containers/podman/v4/pkg/domain/entities"
+	"github.com/containers/podman/v4/pkg/domain/infra"
+	"github.com/containers/storage"
+	"github.com/hashicorp/go-multierror"
+	"github.com/opencontainers/go-digest"
+	v1 "github.com/opencontainers/image-spec/specs-go/v1"
+	"github.com/sirupsen/logrus"
+)
+
+type listBuilder interface {
+	build(ctx context.Context, images map[entities.BuildReport]entities.ImageEngine) (string, error)
+}
+
+type listBuilderOptions struct {
+	cleanup bool
+	iidFile string
+}
+
+type listLocal struct {
+	listName     string
+	storeOptions storage.StoreOptions
+	options      listBuilderOptions
+}
+
+// newLocalManifestListBuilder returns a manifest list builder which saves a
+// manifest list and images to local storage.
+func newLocalManifestListBuilder(listName string, storeOptions *storage.StoreOptions, options listBuilderOptions) (listBuilder, error) {
+	if options.iidFile != "" {
+		return nil, fmt.Errorf("local filesystem doesn't use image IDs, --iidfile not supported")
```

Comment:
The first value returned by the `localEngine.ManifestAdd()` call in `listLocal.build()` returns the value that could be written to this file after a "sha256:" prefix.

Reply:
got it, fixed